infrastructure/repository/userservice/UserServiceSubjectRepository.go

```diff
@@ -144,14 +168,44 @@ func (u *SubjectRepository) GetByOrgID(orgID string) (chan domain.Subject, chan
 	return subChan, errChan
 }
 
+// GetByID retrieves a principal for the given ID. If no ID is provided (ex: empty string), it returns an anonymous principal. If any error occurs, it's returned.
```

Comment:
Do we still want an anonymous principal returned? I'd kind of prefer nil at this point (as implemented below), unless there is a reason.

Reply:
The purpose was to represent a lack of an authenticated principal. Nil could also communicate that, though it may not even matter atm- that's important if some endpoints can be called anonymously, and we need to distinguish which is which in an application service. At the moment, we reject any unauthenticated traffic.